daemon/src/db.rs

```diff
@@ -200,14 +212,34 @@ impl Connection {
     {
         let mut conn = self.inner.acquire().await?;
 
-        let row = load_cfd_row(&mut conn, id).await?;
-        let cfd = C::new(args, row);
+        let cache_key = (TypeId::of::<C>(), id);
+
+        let cfd = match self.aggregate_cache.remove(&cache_key).await {
+            None => {
+                // No cache entry? Load the CFD row. Version will be 0 because we haven't applied
+                // any events, thus all events will be loaded.
+                let cfd = load_cfd_row(&mut conn, id).await?;
+
+                C::new(args, cfd)
+            }
+            Some(cfd) => {
+                // Got a cache entry: Downcast it to the type at hand.
 
-        let cfd = load_cfd_events(&mut conn, id, 0)
+                *cfd.downcast::<C>()
+                    .expect("we index by type id, must be able to downcast")
```

Comment:
❓ Could this casting be removed if we use a `CfdAggregateKind` enum where the variants hold the actual types inside? Maybe it doesn't work or is not worth it.

Reply:
That would require the `db` module to depend on all possible types. At the moment, they all live in the `daemon` crate so it would be possible but really, I think there shouldn't be a need for this dependency.